timelock-impl/src/main/java/com/palantir/atlasdb/timelock/lock/watch/ArrayLockEventSlidingWindow.java

```diff
@@ -44,92 +38,43 @@ long lastVersion() {
-        return ValueAndVersion.of(lastVersion(), task.get());
-    }
-
-    /**
-     * Returns a list of all events that occurred immediately after the requested version up to the most recent version,
-     * ordered by consecutively increasing sequence numbers. If the list cannot be created, either a priori or because
-     * new events are added to the window during execution of this method causing eviction an event before it was read,
-     * the method will return {@link Optional#empty()}.
-     *
-     * Note on concurrency:
-     * 2. Before reading from buffer, we read nextSequence.
-     *
-     * 1. and 2. ensure that calls to this method have an up to date view of buffer, containing all updates made so
-     * far. The buffer may be updated after the volatile read of nextSequence, and these updates may or may not be
-     * visible. This does not affect correctness:
-     *   a) the newer updates are not expected to be reflected in the returned list
-     *   b) if (some of) the newer updates are visible and overwrite a value that should have been included in the
-     *      returned list, it may end up included in the candidate result. This will be detected by
-     *      validateConsistencyOrReturnEmpty, and {@link Optional#empty()} will be returned. This correctly reflects
-     *      the state where, even though all the necessary events were in the requested window at the start of executing
-     *      this method, that is no longer the case when the method returns.
-     */
-    public Optional<List<LockWatchEvent>> getFromVersion(long version) {
-        return getFromTo(version, lastVersion());
-    }
-
-    public Optional<List<LockWatchEvent>> getFromTo(long startVersion, long endVersion) {
-        if (versionInTheFuture(startVersion, endVersion) || versionTooOld(startVersion, endVersion)) {
-            return Optional.empty();
-        }
-
-        int startIndex = LongMath.mod(startVersion + 1, maxSize);
-        int windowSize = Ints.saturatedCast(endVersion - startVersion);
+    public List<LockWatchEvent> getNextEvents(long version) {
+        Preconditions.checkArgument(hasNextEvents(version), "Version not in the log");
+        int startIndex = LongMath.mod(version + 1, maxSize);
+        int windowSize = Ints.saturatedCast(lastVersion() - version);
         List<LockWatchEvent> events = new ArrayList<>(windowSize);
 
         for (int i = startIndex; events.size() < windowSize; i = incrementAndMod(i)) {
             events.add(buffer[i]);
         }
 
-        return validateConsistencyOrReturnEmpty(startVersion, events);
+        return events;
     }
 
     private int incrementAndMod(int num) {
         num++;
         return num >= maxSize ? num % maxSize : num;
     }
 
-    private boolean versionInTheFuture(long lastVersion, long lastWrittenSequence) {
-        return lastVersion > lastWrittenSequence;
+    private boolean validateVersion(long version) {
```

Comment:
This returns true if the version is unloadable, which I find strange. Maybe either flip this, or `isVersionUnknown` or something along those lines.

Reply:
Yeah, I tried not to rewrite this because that's how Grgur wrote these. But you're right, this reads weird.

Reply:
Actually, correction: I rewrote this to be weird, so just gonna fix that.

Reply:
Done